modules/javafx.graphics/src/main/java/com/sun/prism/impl/ps/PaintHelper.java

```diff
@@ -750,6 +750,11 @@ static void setImagePattern(ShaderGraphics g,
         Affine3D at = scratchXform3D;
         g.getPaintShaderTransform(at);
 
+        BaseTransform paintXform = paint.getPatternTransformNoClone();
+        if (paintXform != null) {
```

Comment:
Minor: Is the null check needed ?  I could not find an instance where an object of `ImagePattern` is constructed using default constructor. 
If `ImagePattern.getPatternTransformNoClone()` could return null then should we do null check with other calls to `ImagePattern.getPatternTransformNoClone()` ? OR may be remove this null check.

Reply:
The null check is not actually required as instances of `ImagePattern` can't have a null `patternTransform`. I added this check as `setLinearGradient` and `setRadialGradient` functions have also added them, when instances of `Gradient` can't have null as its transform. Should this null check be removed?

Reply:
I think leaving it for consistency is fine.